x-pack/plugins/ml/public/application/components/annotations/annotations_table/annotations_table.js

```diff
@@ -447,38 +519,119 @@ export class AnnotationsTable extends Component {
       });
     }
 
-    columns.push({
-      align: RIGHT_ALIGNMENT,
-      width: '60px',
-      name: i18n.translate('xpack.ml.annotationsTable.actionsColumnName', {
-        defaultMessage: 'Actions',
-      }),
-      actions,
-    });
-
     const getRowProps = (item) => {
       return {
         onMouseOver: () => this.onMouseOverRow(item),
         onMouseLeave: () => this.onMouseLeaveRow(),
       };
     };
+    let filterOptions = [];
+    if (this.props.aggregations?.event?.buckets) {
+      const buckets = this.props.aggregations.event.buckets;
+      const foundUser = buckets.findIndex((d) => d.key === ANNOTATION_EVENT_USER) > -1;
+      filterOptions = foundUser
+        ? buckets
+        : [{ key: ANNOTATION_EVENT_USER, doc_count: 0 }, ...buckets];
+    }
+    const filters = [
+      {
+        type: 'field_value_selection',
+        field: 'event',
+        name: 'Event',
+        multiSelect: 'or',
+        options: filterOptions.map((field) => ({
+          value: field.key,
+          name: field.key,
+          view: `${field.key} (${field.doc_count})`,
+        })),
+      },
+    ];
+
+    if (
+      this.props.chartDetails?.entityData?.entities &&
+      this.props.chartDetails?.entityData?.entities.length > 0
+    ) {
+      this.props.chartDetails?.entityData?.entities.forEach((entity) => {
+        if (entity.fieldType === 'partition_field') {
+          columns.push({
+            field: 'partition_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.partitionColumnName', {
+              defaultMessage: 'Partition',
+            }),
+            sortable: true,
+          });
+        }
+
+        if (entity.fieldType === 'over_field') {
+          columns.push({
+            field: 'over_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.overColumnName', {
+              defaultMessage: 'Over',
+            }),
+            sortable: true,
+          });
+        }
+        if (entity.fieldType === 'by_field') {
+          columns.push({
+            field: 'by_field_value',
+            name: i18n.translate('xpack.ml.annotationsTable.byColumnName', {
+              defaultMessage: 'By',
+            }),
+            sortable: true,
+          });
+        }
+      });
+
+      filters.push({
```

Comment:
I think I prefer the other option you tried here:

<img width="327" alt="Screen Shot 2020-07-01 at 2 49 56 PM" src="https://user-images.githubusercontent.com/7405507/86339099-19d24080-bc4b-11ea-92e0-927b7b058740.png">

as it's clearer what 'mode' is selected. With the single button, there is no visual indication on the button which mode is active.

Reply:
I updated the Filter for series to be bold when active here https://github.com/elastic/kibana/pull/70198/commits/010c096bab96418150486009d6c09ede5affb21d